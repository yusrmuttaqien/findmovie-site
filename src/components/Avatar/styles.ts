import styled from 'styled-components';

export const Wrapper = styled.figure`
  --_dimensions: 2.5rem;
  background-color: red;
  height: var(--_dimensions);
  width: var(--_dimensions);
  border-radius: 50%;
  overflow: hidden;

  img {
    height: var(--_dimensions);
    width: var(--_dimensions);
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    --_dimensions: 4rem;
  }
`;
