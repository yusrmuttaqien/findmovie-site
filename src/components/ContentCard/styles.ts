import { Link as L } from 'react-router-dom';
import styled from 'styled-components';

interface BackdropProps {
  readonly $placeholder: string[];
}

export const Wrapper = styled.figure`
  padding: 0.75rem 0.8125rem;
  position: relative;
  overflow: hidden;

  #contents {
    position: relative;
    z-index: 2;
    margin-top: 3.375rem;

    #content-subdetail {
      font-size: 0.75rem;
    }

    #content-detail {
      font-size: 0.85rem;
      font-weight: 200;
      display: -webkit-box;
      overflow: hidden;
      text-overflow: ellipsis;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 4;
    }
  }

  &:hover {
    #backdrop-image {
      transform: scale(1.209);
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    padding: 1.5rem 1.6875rem;

    #contents {
      margin-top: 0;

      #content-detail {
        font-size: 0.9375rem;
        -webkit-line-clamp: 3;
        margin-right: 35%;
      }
    }
  }
`;

export const Link = styled(L)`
  font-size: 1.25rem;
  text-decoration: underline;
  font-weight: 600;
  margin-bottom: 0.5rem;
  display: block;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  cursor: ${({ to }) => (to === '#' ? 'default' : 'pointer')};

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    font-size: 1.875rem;
  }
`;

export const Backdrop = styled.div<BackdropProps>`
  z-index: 1;
  position: absolute;
  top: -10px;
  bottom: -10px;
  left: -10px;
  right: -10px;
  background-image: ${({ $placeholder }) => `url(${$placeholder[1]})`};
  background-size: cover;
  background-position: top right;

  img {
    transform: scale(1.1);
    transition: transform 0.3s ease-in-out;
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  &::after {
    content: '';
    position: absolute;
    inset: 0;
    background: linear-gradient(
      360deg,
      rgba(0, 0, 0, 0.91) 0%,
      rgba(0, 0, 0, 0.85) 26.14%,
      rgba(0, 0, 0, 0.71) 46.06%,
      rgba(0, 0, 0, 0.4) 70.73%,
      rgba(217, 217, 217, 0) 100%
    );
    backdrop-filter: blur(2px);
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    background-image: ${({ $placeholder }) => `url(${$placeholder[0]})`};
    background-position: right center;

    &::after {
      background: linear-gradient(
        90deg,
        rgba(0, 0, 0, 0.91) 0%,
        rgba(0, 0, 0, 0.85) 25%,
        rgba(0, 0, 0, 0.71) 50%,
        rgba(0, 0, 0, 0.4) 75%,
        rgba(217, 217, 217, 0) 100%
      );
    }
  }
`;
