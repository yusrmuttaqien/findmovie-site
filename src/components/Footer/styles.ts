import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 1rem 0;
  border-top: 1px solid;
  border-color: ${({ theme }) => theme.colors.white[50]};
  margin-top: 1.25rem;

  p {
    font-size: 0.75rem;
    font-weight: 200;
    text-align: center;
    color: ${({ theme }) => theme.colors.white[70]};
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    margin-top: 2.1875rem;

    p {
      font-size: 1rem;
    }
  }
`;
