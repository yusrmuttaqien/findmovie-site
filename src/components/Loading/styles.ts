import styled from 'styled-components';

export const Wrapper = styled.div`
  overflow: hidden;
  width: 100%;
  height: max-content;

  span {
    display: block;
    width: 100%;
    height: 4px;
    background-color: ${({ theme }) => theme.colors.white[100]};
    animation: stripAnimation 1s infinite linear;
  }

  @keyframes stripAnimation {
    0% {
      transform: translateX(-100%);
    }
    100% {
      transform: translateX(100%);
    }
  }
`;
