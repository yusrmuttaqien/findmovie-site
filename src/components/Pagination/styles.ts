import styled from 'styled-components';

interface ButtonProps {
  readonly $isActive?: boolean;
}

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Button = styled.button<ButtonProps>`
  color: ${({ theme }) => theme.colors.white[50]};
  background-color: transparent;
  border: none;
  cursor: pointer;

  p,
  svg {
    transition: color 0.1s ease-in-out;
  }

  p {
    font-size: 1.0625rem;
  }

  svg {
    font-size: 1.25rem;
  }

  &:last-child {
    flex-direction: row-reverse;

    svg {
      transform: rotate(180deg);
    }
  }

  &:hover {
    p,
    svg {
      color: ${({ theme }) => theme.colors.white[100]};
    }
  }

  &:disabled {
    cursor: no-drop;

    p,
    svg {
      color: ${({ theme, $isActive }) => {
        return $isActive ? theme.colors.white[100] : theme.colors.white[30];
      }};
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    p {
      font-size: 1.75rem;
    }

    svg {
      font-size: 2rem;
    }
  }
`;
