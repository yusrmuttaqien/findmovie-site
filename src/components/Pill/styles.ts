import styled from 'styled-components';

export const Wrapper = styled.figure`
  background-color: ${({ theme }) => theme.colors.white[100]};
  padding: 0.25rem 0.375rem;
  border-radius: 0.25rem;

  p {
    color: ${({ theme }) => theme.colors.black[100]};
    font-size: 0.625rem;
    font-weight: 500;
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    padding: 0.375rem 0.5rem;

    p {
      color: ${({ theme }) => theme.colors.black[100]};
      font-size: 1rem;
    }
  }
`;
