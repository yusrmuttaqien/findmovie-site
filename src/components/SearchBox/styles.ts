import styled from 'styled-components';
import { SearchIcon as SI } from 'assets/svgs';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  width: max-content;
  gap: 1.25rem;

  input {
    width: 11.875rem;
    background-color: transparent;
    border: none;
    color: ${({ theme }) => theme.colors.white[100]};
    font-size: 1.0625rem;
    line-height: 1.5;

    &::placeholder {
      color: ${({ theme }) => theme.colors.white[50]};
    }

    @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
      font-size: 1.3rem;
      width: 15rem;
    }
  }
`;

export const SearchIcon = styled(SI)`
  font-size: 1.25rem;
  cursor: pointer;

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    font-size: 2rem;
  }
`;
