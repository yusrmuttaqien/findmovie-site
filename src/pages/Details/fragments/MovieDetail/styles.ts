import styled from 'styled-components';

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  gap: 1.5625rem;
  padding: 0 ${({ theme }) => theme.space.x.mobile};

  #metadatas {
    h3 {
      font-weight: 500;
      font-size: 1rem;

      &:not([data-meta='title']) {
        margin-bottom: 0.5rem;
      }

      span {
        font-weight: 800;
      }
    }

    > div {
      display: flex;
      flex-wrap: wrap;
      gap: 0.375rem;
    }

    &:last-child p {
      font-size: 0.85rem;
      font-weight: 200;
    }
  }

  @media screen and (min-width: ${({ theme }) => theme.screen.desktop}px) {
    gap: 2rem;
    padding: 0 ${({ theme }) => theme.space.x.desktop};

    #metadatas {
      h3 {
        font-size: 2.125rem;

        &:not([data-meta='title']) {
          margin-bottom: 1rem;
        }
      }

      > div {
        gap: 0.625rem;
      }

      &:last-child p {
        font-size: 1.5rem;
      }
    }
  }
`;
