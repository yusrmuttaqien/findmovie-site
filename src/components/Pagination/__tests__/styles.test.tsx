import { describe, test, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Wrapper, Button } from '../styles';
import theme from 'styles/index';
import 'jest-styled-components';

describe('Pagination component styles', () => {
  test('styled components should match snapshot', () => {
    const { asFragment: fWrapper } = render(<Wrapper theme={theme} />);
    const { asFragment: fButton } = render(<Button theme={theme} />);
    const { asFragment: fButtonActive } = render(<Button $isActive theme={theme} />);

    expect(fWrapper()).toMatchSnapshot();
    expect(fButton()).toMatchSnapshot();
    expect(fButtonActive()).toMatchSnapshot();
  });
});
