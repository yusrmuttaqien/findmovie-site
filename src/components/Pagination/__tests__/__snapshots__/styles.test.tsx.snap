// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination component styles > styled components should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  justify-content: space-between;
}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {

}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Pagination component styles > styled components should match snapshot 2`] = `
<DocumentFragment>
  .c0 {
  color: rgba(199, 199, 199, 0.5);
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.c0 p,
.c0 svg {
  transition: color 0.1s ease-in-out;
}

.c0 p {
  font-size: 1.0625rem;
}

.c0 svg {
  font-size: 1.25rem;
}

.c0:last-child {
  flex-direction: row-reverse;
}

.c0:last-child svg {
  transform: rotate(180deg);
}

.c0:hover p,
.c0:hover svg {
  color: rgb(199, 199, 199);
}

.c0:disabled {
  cursor: no-drop;
}

.c0:disabled p,
.c0:disabled svg {
  color: rgba(199, 199, 199, 0.3);
}

@media screen and (min-width: 1279px) {
  .c0 p {
    font-size: 1.75rem;
  }

  .c0 svg {
    font-size: 2rem;
  }
}

@media screen and (min-width: 1279px) {

}

<button
    class="c0"
  />
</DocumentFragment>
`;

exports[`Pagination component styles > styled components should match snapshot 3`] = `
<DocumentFragment>
  .c0 {
  color: rgba(199, 199, 199, 0.5);
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.c0 p,
.c0 svg {
  transition: color 0.1s ease-in-out;
}

.c0 p {
  font-size: 1.0625rem;
}

.c0 svg {
  font-size: 1.25rem;
}

.c0:last-child {
  flex-direction: row-reverse;
}

.c0:last-child svg {
  transform: rotate(180deg);
}

.c0:hover p,
.c0:hover svg {
  color: rgb(199, 199, 199);
}

.c0:disabled {
  cursor: no-drop;
}

.c0:disabled p,
.c0:disabled svg {
  color: rgb(199, 199, 199);
}

@media screen and (min-width: 1279px) {

}

@media screen and (min-width: 1279px) {
  .c0 p {
    font-size: 1.75rem;
  }

  .c0 svg {
    font-size: 2rem;
  }
}

<button
    class="c0"
  />
</DocumentFragment>
`;
